src/Rings/AbelianClosure.jl

```diff
@@ -526,6 +526,41 @@ end
 (R::QQField)(a::QQAbFieldElem) = R(a.data)
 (R::ZZRing)(a::QQAbFieldElem) = R(a.data)
 
+################################################################################
+#
+#  Conversion to `QQBarFieldElem`
+#
+#  We assume the natural embedding of cyclotomic fields into `QQBarField()`
+#  that is given by the (documented) fact that
+#  `root_of_unity(F::QQBarField, n::Int)` is $\exp(2 \pi i / n)$.
+#
+################################################################################
+
+function (F::QQBarField)(a::QQAbFieldElem)
+  N = a.c
+  cfs = Oscar.coefficients(a.data)
+  r = root_of_unity(F, N)
+  pow = one(F)
+  res = cfs[1] * pow
+  for i in 2:length(cfs)
+    pow = pow * r
+    res = res + cfs[i] * pow
```

Comment:
Could use unsafe ops -- currently this won't make a difference but it will once https://github.com/Nemocas/Nemo.jl/pull/2031 is available.
```suggestion
  tmp = zero(F)
  res = cfs[1] * pow
  for i in 2:length(cfs)
    pow = mul!(pow, r)
    res = addmul!(res, cfs[i], pow, tmp)
```